core/src/main/java/org/commonjava/indy/core/boot/ParallelStreamsBootupAction.java

```diff
@@ -0,0 +1,51 @@
+package org.commonjava.indy.core.boot;
+
+import org.commonjava.indy.action.BootupAction;
+import org.commonjava.indy.action.IndyLifecycleException;
+import org.commonjava.indy.conf.IndyConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+/**
+ * Created by ruhan on 11/22/18.
+ */
+@ApplicationScoped
+public class ParallelStreamsBootupAction
+                implements BootupAction
+{
+    private final Logger logger = LoggerFactory.getLogger( getClass() );
+
+    private final String FORKJOINPOOL_COMMON_PARALLELISM = "java.util.concurrent.ForkJoinPool.common.parallelism";
+
+    @Inject
+    private IndyConfiguration config;
+
+    @Override
+    public String getId()
+    {
+        return "parallel streams common pool";
+    }
+
+    @Override
+    public void init() throws IndyLifecycleException
+    {
+        if ( System.getProperty( FORKJOINPOOL_COMMON_PARALLELISM ) == null )
+        {
+            if ( config.getForkJoinPoolCommonParallelism() > 0 )
+            {
+                String value = new Integer( config.getForkJoinPoolCommonParallelism() ).toString();
+                System.setProperty( FORKJOINPOOL_COMMON_PARALLELISM, value );
+                logger.info( "Set {}={}", FORKJOINPOOL_COMMON_PARALLELISM, value );
```

Comment:
It would be good to pull the value from the system properties as a check, since if that value is already set (on the command line, etc), I believe this logic will NOT reset it.

Reply:
I thought if the user set it via command line (JAVA_OPTS) then we would honor it in the first. Do you mean if the value in main.conf should take higher priority in this case? Or we just set the larger one?